File: tags/CSLA-2.0.0-Samples/CSharp/PetShop/Source/PetShop.UI/Controls/ShoppingCartControl.ascx.cs
using System;
using System.Web.UI;
using System.Web.UI.WebControls;

using PetShop.Business;
using PetShop.Controls;

namespace PetShop.UI.Controls
{
    public partial class ShoppingCartControl : UserControl
    {
        /// <summary>
        /// Handle Page load event
        /// </summary>
        protected void Page_PreRender(object sender, EventArgs e)
        {
            if (!IsPostBack)
            {
                BindCart();
            }
        }

        /// <summary>
        /// Bind repeater to Cart object in Profile
        /// </summary>
        private void BindCart()
        {
            Profile profile = ProfileManager.Instance.GetCurrentUser(Page.User.Identity.Name);
            
            Business.CartList items = profile.ShoppingCart;
            if (items.Count > 0)
            {
                repShoppingCart.DataSource = items;
                repShoppingCart.DataBind();
                PrintTotal();
                plhTotal.Visible = true;
            }
            else
            {
                repShoppingCart.Visible = false;
                plhTotal.Visible = false;
                lblMsg.Text = "Your cart is empty.";
            }
        }

        /// <summary>
        /// Recalculate the total
        /// </summary>
        private void PrintTotal()
        {
            Profile profile = ProfileManager.Instance.GetCurrentUser(Page.User.Identity.Name);
            if (profile.ShoppingCart.Count > 0)
                ltlTotal.Text = profile.ShoppingCart.Total.ToString("c");
        }

        /// <summary>
        /// Calculate total
        /// </summary>
        protected void BtnTotal_Click(object sender, ImageClickEventArgs e)
        {
            Profile profile = ProfileManager.Instance.GetCurrentUser(Page.User.Identity.Name);
            
            foreach (RepeaterItem row in repShoppingCart.Items)
            {
                TextBox txtQuantity = (TextBox) row.FindControl("txtQuantity");
                ImageButton btnDelete = (ImageButton) row.FindControl("btnDelete");
                
                int quantity;
                if (int.TryParse(WebUtility.InputText(txtQuantity.Text, 10), out quantity))
                {

                    if (quantity > 0)
                        profile.ShoppingCart.SetQuantity(btnDelete.CommandArgument, quantity);
                    else if (quantity == 0)
                        profile.ShoppingCart.Remove(btnDelete.CommandArgument);
                }
            }

            profile = profile.Save();

            BindCart();
        }

        /// <summary>
        /// Handler for Delete/Move buttons
        /// </summary>
        protected void CartItem_Command(object sender, CommandEventArgs e)
        {
            Profile profile = ProfileManager.Instance.GetCurrentUser(Page.User.Identity.Name);
            switch (e.CommandName)
            {
                case "Del":
                    profile.ShoppingCart.Remove(e.CommandArgument.ToString());
                    break;
                case "Move":
                    profile.ShoppingCart.Remove(e.CommandArgument.ToString());
                    profile.WishList.Add(e.CommandArgument.ToString(), profile.UniqueID, false);
                    break;
            }

            profile = profile.Save();
         
            BindCart();
        }
    }
}

File: tags/CSLA-4.0-Samples/VisualBasic/PetShop/Source/PetShop.UI/Controls/ShoppingCartControl.ascx.vb
Imports PetShop.Business
Imports PetShop.Controls


Partial Public Class ShoppingCartControl
    Inherits System.Web.UI.UserControl

    ''' <summary>
    ''' Handle Page load event
    ''' </summary>
    Protected Sub Page_PreRender(ByVal sender As Object, ByVal e As EventArgs)
        If Not IsPostBack Then
            BindCart()
        End If
    End Sub

    ''' <summary>
    ''' Bind repeater to Cart object in Profile
    ''' </summary>
    Private Sub BindCart()
        Dim profile As Profile = ProfileManager.Instance.GetCurrentUser(Page.User.Identity.Name)

        Dim items As Business.CartList = profile.ShoppingCart
        If items.Count > 0 Then
            repShoppingCart.DataSource = items
            repShoppingCart.DataBind()
            PrintTotal()
            plhTotal.Visible = True
        Else
            repShoppingCart.Visible = False
            plhTotal.Visible = False
            lblMsg.Text = "Your cart is empty."
        End If
    End Sub

    ''' <summary>
    ''' Recalculate the total
    ''' </summary>
    Private Sub PrintTotal()
        Dim profile As Profile = ProfileManager.Instance.GetCurrentUser(Page.User.Identity.Name)
        If profile.ShoppingCart.Count > 0 Then
            ltlTotal.Text = profile.ShoppingCart.Total.ToString("c")
        End If
    End Sub

    ''' <summary>
    ''' Calculate total
    ''' </summary>
    Protected Sub BtnTotal_Click(ByVal sender As Object, ByVal e As ImageClickEventArgs)
        Dim profile As Profile = ProfileManager.Instance.GetCurrentUser(Page.User.Identity.Name)

        For Each row As RepeaterItem In repShoppingCart.Items
            Dim txtQuantity As TextBox = DirectCast(row.FindControl("txtQuantity"), TextBox)
            Dim btnDelete As ImageButton = DirectCast(row.FindControl("btnDelete"), ImageButton)

            Dim quantity As Integer
            If Integer.TryParse(WebUtility.InputText(txtQuantity.Text, 10), quantity) Then

                If quantity > 0 Then
                    profile.ShoppingCart.SetQuantity(btnDelete.CommandArgument, quantity)
                ElseIf quantity = 0 Then
                    profile.ShoppingCart.Remove(btnDelete.CommandArgument)
                End If
            End If
        Next

        profile = profile.Save(True)

        BindCart()
    End Sub

    ''' <summary>
    ''' Handler for Delete/Move buttons
    ''' </summary>
    Protected Sub CartItem_Command(ByVal sender As Object, ByVal e As CommandEventArgs)
        Dim profile As Profile = ProfileManager.Instance.GetCurrentUser(Page.User.Identity.Name)
        Select Case e.CommandName
            Case "Del"
                profile.ShoppingCart.Remove(e.CommandArgument.ToString())
                Exit Select
            Case "Move"
                profile.ShoppingCart.Remove(e.CommandArgument.ToString())
                profile.WishList.Add(e.CommandArgument.ToString(), profile.UniqueID, False)
                Exit Select
        End Select

        profile = profile.Save(True)

        BindCart()
    End Sub

End Class